react/features/calendar-sync/components/ConferenceNotification.native.js

```diff
@@ -0,0 +1,243 @@
+                                <Icon
+                                    name = 'navigate_next'
+                                    style = { styles.notificationIcon } />
+                            </View>
+                        </View>
+                    </TouchableOpacity>
+                </View>
+            );
+        }
+
+        return null;
+    }
+
+    _getContainerStyle: () => Array<Object>;
+
+    /**
+     * Assembles a style array for the container.
+     *
+     * @private
+     * @returns {Array<Object>}
+     */
+    _getContainerStyle() {
+        const { event } = this.state;
+        const now = Date.now();
+        const style = [
+            styles.notificationContainer,
+            {
+                top: this.props._screenHeight / 10
+            }
+        ];
+
+        if (event && event.startDate < now && event.endDate > now) {
+            style.push(styles.notificationContainerPast);
+        }
+
+        return style;
+    }
+
+    _maybeDisplayNotification: () => void;
+
+    /**
+     * Periodically checks if there is an event in the calendar for which we
+     * need to show a notification.
+     *
+     * @private
+     * @returns {void}
+     */
+    _maybeDisplayNotification() {
+        const { _currentConferenceURL, _eventList } = this.props;
+        let eventToShow;
+
+        if (_eventList && _eventList.length) {
+            const now = Date.now();
+
+            for (const event of _eventList) {
+                if (event.url !== _currentConferenceURL) {
+                    if ((!eventToShow
+                        && event.startDate > now
+                        && event.startDate < now + ALERT_MILLISECONDS)
+                        || (event.startDate < now && event.endDate > now)
+                    ) {
+                        eventToShow = event;
+                    }
+                }
+            }
+        }
+
+        this.setState({
+            event: eventToShow
+        });
+    }
+
+    _onGoToNext: () => void;
+
+    /**
+     * Opens the meeting URL that the notification shows.
+     *
+     * @private
+     * @param {string} url - The URL to open.
+     * @returns {void}
+     */
+    _onGoToNext() {
+        const { event } = this.state;
+
+        if (event && event.url) {
+            this.props.dispatch(appNavigate(event.url));
+        }
+    }
+
+}
+
+/**
+ * Maps redux state to component props.
+ *
+ * @param {Object} state - The redux state.
+ * @returns {{
+ *      _eventList: Array,
+ *      _screenHeight: number
+ * }}
+ */
+export function _mapStateToProps(state: Object) {
+    const { height } = Dimensions.get('window');
+    const { locationURL } = state['features/base/connection'];
+
+    return {
+        _currentConferenceURL:
+            locationURL ? getURLWithoutParams(locationURL)._url : '',
+        _eventList: state['features/calendar-sync'].events,
+        _screenHeight: height
+    };
+}
```

Comment:
What about a separate object that's in charge of calendar events and does its polling and then this view is only responsible for showing based on the presence of a notification in the redux store? That separates a lot of logic from this component and the component becomes focused on display only. Maybe that's too much re-writing though.

Reply:
You mean like extracting the "_maybeDisplayNotification" code into something, or you mean more different approach? What would trigger that object to poll/check if something is to be shown?

Reply:
I suppose the top level view somewhere would trigger the object's olling. The object could take in a callback or dispatch (probably a callback to keep it more isolated from its surroundings). I'm not fighting for this change. I've already cooled from it.